packages/accounts/src/light-account/account.ts

```diff
@@ -66,25 +73,23 @@ export type CreateLightAccountParams<
 } & EntryPointParameter<TEntryPointVersion>;
 
 export async function createLightAccount<
-  TEntryPointDef extends EntryPointDefRegistry[EntryPointVersion],
-  TEntryPointVersion extends EntryPointVersion = TEntryPointDef extends EntryPointDef<
-    infer U
-  >
-    ? U
-    : never,
+  TEntryPointVersion extends DefaultEntryPointVersion = DefaultEntryPointVersion,
```

Comment:
This is needed, I don't think we should require devs to specify the entry point version, and this is only one of the overloaded function definition for createLightAccount.
If you look at the function definition that contains implementation, it starts with TTransport which is backward compatible.

Reply:
if you want to avoid casting in the final overload definition, keep the order + number of generics the same across all the overloads with the permissive union types.

The old implementation of `createLightAccount` only had 2 overloads (we can have more), but the first specified the union type of generics + what should be extended. The final overload used the default values for all of the generics. In https://github.com/alchemyplatform/aa-sdk/pull/544 the result of doing this is that we don't have to typecast within the final overload because the defaults support union typing. The final output however, will be specialized based on the various input params